py-polars/polars/utils/_parse_expr_input.py

```diff
@@ -89,6 +171,8 @@ def parse_as_expression(
         strings are parsed as column names.
     structify
         Convert multi-column expressions to a single struct expression.
```

Comment:
No strong opinion on my side. Though I wonder, is this truly simpler? I never like UNION output types much.

Reply:
True, it's a bit meh... maybe we can have another pair of utils for pyexpr / expr:

* `parse_as_expression`
* `parse_as_pyexpression`
* `parse_as_list_of_expressions`
* `parse_as_list_of_pyexpressions`